### PR TITLE
[MRG+1] Fixes Bug for Rendering PDF Legends with Raster and Vector Objects

### PR DESCRIPTION
**Pull request by ColinDruery**
## PR Summary

This addition to the DrawingArea class implements a check for the case of rendering a PDF with a legend containing both raster and vector objects. If these requirements are met I un-rasterized the child elements allowing the PDF to render with the expected results. I have provided a few test cases and baseline images to compare to. These test cases cover: only one element being rasterized, more than one element being rasterized, and the entire legend being rasterized. This change should be non impactful to current functionality as it is only executed when these conditions are met. To verify this change didn't cause issues with the same functionality on a png, the test cases run for both pdf and png.

This PR is related to issue https://github.com/matplotlib/matplotlib/issues/6227
## PR Checklist
- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_offsetbox.py [hunk 1]
```diff
@@ -99,3 +99,137 @@ def test_offsetbox_loc_codes():
         anchored_box = AnchoredOffsetbox(loc=code, child=da)
         ax.add_artist(anchored_box)
     fig.canvas.draw()
+
+
+@image_comparison(baseline_images=['rasterized_patch'],
+                  extensions=['pdf'], tol=0.01)
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_offsetbox.py]: You can list multiple extensions `extensions=['pdf', 'png']` to reduce the number of tests you need to add.

**Comment by tacaswell** [comment]
This does not seem like the right fix as it just band-aids over the real problem (which is something not going right in the offset box stack) rather than actually fixing the bug.

File: lib/matplotlib/tests/test_offsetbox.py (hunk 1 above)
Comment (ColinDruery) [inline lib/matplotlib/tests/test_offsetbox.py]: Thank you for the feedback on the tests, I will make these updates.

**Comment by ColinDruery** [comment]
Yes I agree with you, a more suitable title for this PR would be "workaround" rather than "Fixes". I had just found this way to disallow the problem behaviour and thought I would purpose it to get thoughts on if this would be helpful for the time between now and when a larger refactor to completely fix the issue can be implemented.

**Comment by tacaswell** [comment]
As a middle ground, how about adding a comment to one of the doc-pages about legend (maybe in https://github.com/matplotlib/matplotlib/blob/master/doc/users/legend_guide.rst ) identifying this as a known issue with a for-loop that will go through the children of a legend and turn rasterization off on them?

**Comment by ColinDruery** [comment]
Please review my changes to legend_guide.rst when you have a chance to see if you think I have made the addition in the appropriate place, as well as provided the correct amount of information.

**Comment by QuLogic** [comment]
I think @tacaswell's middle ground is _just_ the documentation with the for-loop, and all of the other changes should be dropped.

**Comment by jklymak** [comment]
I'll close as abandoned.  Thanks for the PR though, sorry it didn't get through!

**Pull request closed without merging**